node_modules.py

```diff
@@ -43,13 +45,144 @@
 OBS_SCM_COMPRESSION = None
 
 
-def collect_deps_recursive(d, deps):
+class CpioReader(object):
+    def __init__(self, fn):
+        self.fh = open(fn, 'rb')
+
+    def extract(self, outdir):
+
+        class CpioFile(object):
```

Comment:
I think that if the class live at the upper level will make the method more clear, even if the class is used only here.